samples/build-duration-logger/index.js

```diff
@@ -0,0 +1,255 @@
+
+            if (Object.getOwnPropertyNames(handlerClass.prototype).includes('complete')) {
+                addHandler('complete', handler);
+            }
+
+            return handlers;
+        }, {});
+    }
+
+    processBuild(build) {
+        this.numBuildsInProcess++;
+        const buildEventHandlers = this.createBuildEventHandlers(build);
+        const buildEventStreamUrl = this.createBuildEventStreamUrl(build.buildId);
+
+        createServerSideEventStream(buildEventStreamUrl, {
+            oncomplete:  () => {
+                // there isn't an onclose event that we can listen to, but an error event does get triggered
+                // when the end of the stream is reached so use that as a rough proxy for the end of the stream
+                this.finishedProcessingBuild();
+
+                // Call the 'complete()' method on any handler that has it.
+                if (buildEventHandlers.complete) {
+                    buildEventHandlers.complete.forEach(handler => handler.complete());
+                }
+            },
+            eventListeners: [
+                {
+                    eventName: 'BuildEvent',
+                    eventHandler:event => {
+                        const buildEventPayload = JSON.parse(event.data);
+                        const { eventType } = buildEventPayload.type;
+
+                        if (this.allHandledEventTypes.has(eventType)) {
+                            buildEventHandlers[eventType].forEach(handler => handler[`on${eventType}`](buildEventPayload));
+                        }
+                    }
+                }
+            ],
+            retry: {
+                interval: 2000,
+                maxRetries: 100
+            }
+        });
+    }
+
+    finishedProcessingBuild() {
+        this.numBuildsInProcess--;
+        setTimeout(() => this.processPendingBuilds(), 0); // process the next set of pending builds, if any
+    }
+}
+
+function createServerSideEventStream(url, configuration) {
+    const STATUS_COMPLETE = 204;
+
+    let stream;
+    let retries;
+
+    const _onopen = configuration.onopen || noop;
+    const _onerror = configuration.onerror || noop;
+    const _oncomplete = configuration.oncomplete || noop;
+    const _configurationRetry = configuration.retry || { }
+    const _maxRetries = _configurationRetry.maxRetries || 0;
+    const _retryInterval = _configurationRetry.interval || 1000;
+
+    stream = createStream()
+
+    function createStream() {
+        stream = new EventSourcePolyfill(url, { headers: {'Authorization': `Basic ${BASIC_AUTH_TOKEN}`}});
+
+        stream.reconnectInterval = _retryInterval;
+
+        stream.onopen = (event) => {
+            retries = 0;
+            _onopen(event)
+        }
+
+        stream.onerror = (event) => {
+            // the server will sent a 204 status when closing the stream if no more content is present but the
```

Comment:
Can you please update the comments to what I proposed on the old comments? These are all the old comments.

Reply:
Looks like I messed up using both the Github UI for accepting requested changes and IDEA, the changes were at some point accepted but now they are gone. I'll add you changes back.

Reply:
Added back.